File: src/main/php/com_github_leetcode/ListNode.php
<?php

namespace leetcode\com_github_leetcode;

class ListNode {
    public $val;
    public $next;

    public function __construct($val = 0, $next = null) {
        $this->val = $val;
        $this->next = $next;
    }

    public function toString() {
        $result = (string) $this->val;

        if ($this->next !== null) {
            $current = $this->next;

            while ($current->next !== null) {
                $result .= ', ' . $current->val;
                $current = $current->next;
            }

            $result .= ', ' . $current->val;
        }

        return $result;
    }
}

File: src/main/scala/com_github_leetcode/ListNode.scala
package com_github_leetcode

class ListNode(_x: Int = 0, _next: ListNode = null) {
    var next: ListNode = _next
    var x: Int = _x

    override def toString: String = {
        val result = new StringBuilder(x.toString)
        var current = next
        while (current != null) {
            result.append(", ")
            result.append(current.x)
            current = current.next
        }
        result.toString
    }
}
